controllers/configuration_controller.go

```diff
@@ -182,11 +182,21 @@ func (r *ConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			}
 		}
 
-		if err := r.terraformDestroy(ctx, configuration, meta); err != nil {
-			if err.Error() == types.MessageDestroyJobNotCompleted {
-				return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
+		// fix: The failure of terraformApply does not mean that it has never been successful,
+		//     but if the 'tfstate' file does not exist, then there is no need for terraformDestroy,
+		//     which can quickly clean up resources.
+		if meta.isTFStateGenerated(ctx) {
+			if err := r.terraformDestroy(ctx, configuration, meta); err != nil {
+				if err.Error() == types.MessageDestroyJobNotCompleted {
+					return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
+				}
+				return ctrl.Result{RequeueAfter: 3 * time.Second}, errors.Wrap(err, "continue reconciling to destroy cloud resource")
+			}
+		} else {
+			klog.Infof("No need to execute terraform destroy command, because tfstate file not found: %s/%s", configuration.Namespace, configuration.Name)
+			if err := r.cleanUpSubResources(ctx, configuration, meta); err != nil {
+				klog.Warningf("Failed to clean up sub-resources, but it's ignored as the resources are being forced to delete: %s", err)
```

Comment:
No clue suggests the "configuration" is force to delete. 
```suggestion
				klog.Warningf("Ignoring error when clean up sub-resources, for no resource is actually created: %s", err)
```